### PR TITLE
feat(ci): migrate sentry to gha

### LINKED ISSUE #10644: migrate circle ci sentry job to gha agw build job on push

### PR DESCRIPTION
**Pull request by quentinDERORY**
## Summary

Migrating sentry to gha and will be kicked as part of every agw build done on master and new releases.

## Test Plan

Tested on my forked master

## Additional Information

For releases we still need to find a way to dynamically adjust the version (currently being 1.7.0)

### CODE REVIEW

File: .github/workflows/build_all.yml [hunk 1]
```diff
@@ -159,12 +160,24 @@ jobs:
...
+        if: github.event_name == 'push'
+        with:
+          name: sentry-exec
+          path: lte/gateway/magma-packages
+      - name: Extract magma version
+        # yamllint enable
+        if: github.event_name == 'push'
+        id: magma_version
+        run: |
+          MAGMA_VERSION=ls lte/gateway/magma-packages/ | grep "^magma_[0-9].*"xargs -I "%" dpkg -I ~/magma-packages/% | grep "Version" | awk \'{print $2}\'
```
Comment (jheidbrink) [inline .github/workflows/build_all.yml]: #10832  changes this for CircleCI.

File: .github/workflows/composite/sentry-create-and-upload-artifacts/action.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        # in $MAGMA_ROOT/circleci in the magma_integ_test step
+        cd "${{ inputs.PATH_TO_EXEC }}"
+
+        objcopy --only-keep-debug "${{ inputs.EXECUTABLE_NAME }}" "{{ inputs.EXECUTABLE_NAME }}".debug
+        objcopy --strip-debug --strip-unneeded "{{ inputs.EXECUTABLE_NAME }}"
+        objcopy --add-gnu-debuglink="{{ inputs.EXECUTABLE_NAME }}".debug "{{ inputs.EXECUTABLE_NAME }}"
+    - name: Upload debug artifacts for an executable to Sentry.io
+      env:
+        executable_name: sessiond
+        projects: ["lab-agws-native", "radtonics_lab_native", "magma-staging-native", "si-agw-native"]
```
Comment (themarwhal) [inline .github/workflows/composite/sentry-create-and-upload-artifacts/action.yml]: As a note for maybe future improvements, sentry does support just uploading to a custom repository (https://docs.sentry.io/platforms/native/data-management/debug-files/symbol-servers/) . Then maybe we don't have to upload to every new project we add, we can just add it to one place. Maybe I'll add an issue to file this :D

**Review by themarwhal — APPROVED** [review]

File: .github/workflows/build_all.yml [hunk 3]
```diff
@@ -191,6 +196,50 @@ jobs:
...
+          SENTRY_ORG: lf-9c
+          PATH_TO_EXEC: "./executables"
+          SENTRY_AUTH_TOKEN: "${{ secrets.SENTRY_AUTH_TOKEN }}"
+      - name: Create a release in Sentry.io with the commit hash
+        env:
+          SENTRY_AUTH_TOKEN: "${{ secrets.SENTRY_AUTH_TOKEN }}"
+        run: |
+           # The assumption here is that the magma version is copied out
+           # into /home/circleci/project/circleci in the magma_integ_test step
+           COMMIT_HASH_WITH_VERSION="magma@1.7.0.$(git rev-list --count HEAD)-${GITHUB_SHA:0:9}"
```
Comment (jheidbrink) [inline .github/workflows/build_all.yml]: I think we should use `${GITHUB_SHA:0:8}` here to align with the `get_commit_hash` [function](https://github.com/magma/magma/blob/master/orc8r/tools/fab/pkg.py#L65-L66) that is used to [determine the version](https://github.com/magma/magma/blob/9e9cb764e57d038fa9d8a08d81b49a3ddc526aa8/lte/gateway/release/build-magma.sh#L268) in `build-magma-sh`.

**Review by jheidbrink — CHANGES_REQUESTED** [review]

File: .github/workflows/build_all.yml (hunk 3 above)
Comment (jheidbrink) [reply .github/workflows/build_all.yml]: Now that we don't read `magma_version` anymore, we can remove the comment above.

**Review by jheidbrink — APPROVED** [review]

**Pull request merged**